### PR TITLE
Allow deletion of external managed cks nodes

### PR DESCRIPTION
**Pull request by vishesh92**
### Description

The user might create the cluster manually. And since the cluster is being managed by some external provider, it's possible that provider doesn't have context of CKS and deletes the VM. This can lead to failures.

This PR allows deletion of CKS node if the CKS cluster is Externally managed.

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)
- [ ] build/CI

#### Feature/Enhancement Scale

- [ ] Major
- [x] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [x] Minor
- [ ] Trivial

### CODE REVIEW

**Review by weizhouapache — APPROVED** [review]
code lgtm

**Comment by vishesh92** [comment]
@blueorangutan package

File: plugins/integrations/kubernetes-service/src/main/java/com/cloud/kubernetes/cluster/KubernetesClusterHelperImpl.java [hunk 1]
```diff
@@ -58,6 +58,9 @@ public void checkVmCanBeDestroyed(UserVm userVm) {
         KubernetesCluster kubernetesCluster = kubernetesClusterDao.findById(vmMapVO.getClusterId());
         String msg = "Instance is a part of a Kubernetes cluster";
         if (kubernetesCluster != null) {
+            if (kubernetesCluster.getClusterType().equals(KubernetesCluster.ClusterType.ExternalManaged)) {
```
Comment (shwstppr) [inline plugins/integrations/kubernetes-service/src/main/java/com/cloud/kubernetes/cluster/KubernetesClusterHelperImpl.java:61]:
```suggestion
            if (KubernetesCluster.ClusterType.ExternalManaged.equals(kubernetesCluster.getClusterType())) {
```
check should be safer other way around

**Review by shwstppr — COMMENTED** [review]
What happens to the cluster-vm mapping? Do we need to remove it?

**Comment by vishesh92** [comment]
> What happens to the cluster-vm mapping? Do we need to remove it?

I don't think we need to remove it. Worst case an entry is left over in the table.

**Review by kiranchavala — APPROVED** [review]
LGTM, steps to test the issue

1. Launch a cks cluster
2. Launch a capc cluster
3. Create a externally managed cluster with API call 

create kubernetescluster clustertype=ExternalManaged name=<> zoneid<>

4. Add the capc based control and worker nodes to the cluster

 (cmk) > add virtualmachinestokubernetescluster id=<> virtualmachineids=<> 

5. Try to delete the cks node , exception is thrown



Screenshot 2024-06-14 at 2 21 26 PM

6. Only the delete/expunge of the capc node is allowed
Screenshot 2024-06-14 at 2 44 26 PM

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by rohityadavcloud** [comment]
@blueorangutan test

**Pull request merged**